templatefunctions/striptags_func.go

```diff
@@ -2,42 +2,132 @@ package templatefunctions
 
 import (
 	"context"
+	"regexp"
 	"strings"
 
+	"flamingo.me/flamingo/v3/framework/config"
 	"golang.org/x/net/html"
 )
 
 type (
-	StriptagsFunc struct{}
+	// StriptagsFunc provides template function to strip html tags
+	StriptagsFunc     struct{}
+	allowedAttributes map[string]struct{}
+	allowedTags       map[string]allowedTag
+	allowedTag        struct {
+		name       string
+		attributes allowedAttributes
+	}
+)
+
+var (
+	nameRe          = regexp.MustCompile(`[a-z0-9\-]+`)
+	attributesRe    = regexp.MustCompile(`[a-z0-9]+([a-z]+)`)
```

Comment:
This does not conform with allowed HTML attribute names, e.g. including `-` (such as `data-qa`) and `:` (such as `v-bind:class`).

Reply:
Okay, how should we handle non valid HTML attributes? According to https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes v-bind:class isn't valid..

I guess we have to find the balance between sticking to the HTML specification and our common use cases.. I will add a more general RegEx that does not comply with the HTML standard but fulfils your test cases..

Reply:
https://html.spec.whatwg.org/multipage/syntax.html#syntax-attribute-name it is valid? ;)